### ISSUE TITLE
Elasticsearch Citation Search seems broken

### ISSUE THREAD

**Issue by flooie** [freelawproject/courtlistener#4633]
https://www.courtlistener.com/opinion/595912/go/#fn-s_ref

When querying for the preceding case in Elastic using the citation, I receive the following message:

	Showing results for “” without citation “980 F.2d 737”. It appears we don’t yet have that citation.

However, filtering by a portion of the case name confirms that both the case and citation are indeed present in the system, as shown below:

```
Ann Franco v. National Van Lines, Inc. Starving Students, Inc., Ann Franco, and 
Louis G. Fazzi, Esq. v. National Van Lines, Inc. Starving Students, Inc., 
980 F.2d 737 (9th Cir. 1992)
Court of Appeals for the Ninth Circuit

Filed: November 13th, 1992
Precedential Status: Non-Precedential
Citations: 980 F.2d 737, 1992 U.S. App. LEXIS 35503
Docket Number: 89-55697
```
https://www.courtlistener.com/?q=980%20F.2d%20737&type=o&order_by=score%20desc&case_name=Ann%20Franco%20v.%20National%20Van%20Lines&stat_Published=on&stat_Unpublished=on

![Image](https://github.com/user-attachments/assets/b1e5a8d3-7413-4254-9747-5c36323df0b8)

It appears there may be an issue with citation-based retrieval in Elastic.

**Comment by mlissner** [comment]
OK, yeah, a couple things weird about that, eh? 

1. It shouldn't eliminate the citation from the query and return "" when the only thing there is the citation.

1. But did it do that? Somehow it still returned results.

1. Looking it up by citation works: 

https://www.courtlistener.com/?q=&type=o&order_by=score%20desc&case_name=Ann%20Franco%20v.%20National%20Van%20Lines&stat_Published=on&stat_Unpublished=on&citation=980%20F.2d%20737

1. But like this does not:

https://www.courtlistener.com/?q=citation%3A(980%20F.2d%20737)&type=o&order_by=score%20desc&case_name=Ann%20Franco%20v.%20National%20Van%20Lines&stat_Published=on&stat_Unpublished=on

1. But, lo, this works (with the space moved):

https://www.courtlistener.com/?q=citation%3A(980%20F.%202d%20737)&type=o&order_by=score%20desc&case_name=Ann%20Franco%20v.%20National%20Van%20Lines&stat_Published=on&stat_Unpublished=on

I think the whitespace is the heart of the issue here, but we can put this on Alberto's backlog to dig in.

**Comment by mlissner** [comment]
Got another instance of this today where whitespace doesn't seem to be the issue: 

https://www.courtlistener.com/?q=101%20So.%20844&type=o&order_by=score%20desc&stat_Published=on

![Image](https://github.com/user-attachments/assets/df370014-c852-43f0-93c9-aa0977a5cd7f)

But `101 So. 844` does exist: 

https://www.courtlistener.com/opinion/3239863/u-s-salvage-sales-co-v-weber/

As we're fixing this, Rebecca dropped this meme into our Slack: 

![Image](https://github.com/user-attachments/assets/c13ec3ec-f833-4596-8963-febeed42bfe8)

So maybe if it makes sense, we should also make pure citation searches just take people where they want to go when we do this (if it doesn't add too much complexity or time).

**Comment by mlissner** [comment]
@flooie I think it might be better if y'all worked on this one, since I have a feeling it might relate to random citation parsing idiosyncrasies.

**Comment by flooie** [comment]
Got it.

It finally dawned on me that all the examples were for unpublished opinions.

I flipped this code:

```
if not hasattr(full_citation, "citing_opinion"):
    full_citation.citing_opinion = None

search_query = OpinionDocument.search()
filters = [
    Q("term", **{"status.raw": "Unpublished"}),  # Non-precedential documents aren't cited
    Q("match", cluster_child="opinion"),
]
```
Setting the status filter to "Unpublished" fixed the issue locally, which makes sense.

⸻

Next, I ran this query on the server to see how many opinion clusters are marked as something other than “Published” but have citations:

```
from cl.search.models import OpinionCluster

clusters = OpinionCluster.objects.exclude(
    precedential_status="Published"
).filter(
    citations__isnull=False
).distinct()

clusters.count()
# Out: 429,468
```

So, it turns out we have 429,468 clusters with citations that are not marked as published. The simplest answer might be to just mark them all as published, but that wouldnt be correct either.  @mlissner.

⸻

Looking a bit deeper:

Federal Appendix (F. App’x)

Most of the first tranche are citations to the Federal Appendix:

```
Unpublished <QuerySet [<Citation: 373 F. App'x 269>]>
Unpublished <QuerySet [<Citation: 373 F. App'x 172>]>
```


According to [Wikipedia](https://en.wikipedia.org/wiki/Federal_Appendix):

The Federal Appendix was published from 2001 to 2021. It contains opinions not designated for publication. After Rule 32.1 (eff. 2007), such opinions can be cited, but they’re still not treated as binding precedent. 

So while these aren’t “precedential,” they are citeable.  which again puts us in a pickle.

The next batch here just seem incorrectly labeled.  

Federal Reporter (F.3d and F.2d)

Later entries include F.3d and F.2d citations:

```
<QuerySet [<Citation: 132 F.3d 1453>]>
Unpublished <QuerySet [<Citation: 131 F.3d 140>]>
Unpublished <QuerySet [<Citation: 980 F.2d 742>]>
...
```

These should be published. That these are marked unpublished is likely a misclassification.

⸻

State-Specific Examples

Then we have things like:

Connecticut: (maybe published maybe not? I dont know)
```
Unpublished <QuerySet [<Citation: 1997 Conn. Super. Ct. 8272>]>
```
Illinois: Seems like its designated specifically unpublished
```
Unpublished <QuerySet [<Citation: 2024 IL App (1st) 211175-U>]>
```
New York Slip Opinions: Same here.
```
Unpublished <QuerySet [<Citation: 2024 NY Slip Op 50918(U)>]>
```

These are officially designated as “unpublished” in their respective systems. They’re often citeable, but not precedential, depending on the state rules. They’re a gray area: not precedential, but arguably “published enough” for search and citation.

@mlissner What would like?

**Comment by mlissner** [comment]
I think two things:

1. The fix for the immediate issue might be easy? Just search for non-precedential stuff when people put citations in the main query box. Remove that optimization.

2. The fix for the larger issue is complicated. Probably the easiest thing is to showing only precedential stuff by default, and to clean up the ones that are messy. That might give us the best of both worlds: The labels are accurate, and the search returns what you might expect.

**Comment by mlissner** [comment]
Oops, forgot to flag you, @flooie ☝🏻

**Comment by flooie** [comment]
uh..... or perhaps use the filter for precedential stuff or non published stuff to be used in the query?

**Comment by mlissner** [comment]
Yeah, that's what I meant. Should we create a second issue to make non-p and precedential stuff appear in results by default?
